flake.nix

```diff
@@ -0,0 +1,84 @@
+{
+  inputs = {
+    systems.url = "github:nix-systems/default";
+    nixpkgs.url = "github:nixos/nixpkgs/nixpkgs-unstable";
+  };
+
+  outputs =
+    {
+      self,
+      nixpkgs,
+      systems,
+    }:
+    let
+      forEachSystem =
+        f: nixpkgs.lib.genAttrs (import systems) (system: f { pkgs = import nixpkgs { inherit system; }; });
+    in
+    {
+      devShells = forEachSystem (
+        { pkgs }:
+        {
+          default =
+            pkgs.mkShell.override
+              {
+                # override clang 16 runtime
+                stdenv = pkgs.lowPrio pkgs.llvmPackages_16.stdenv;
+              }
+              {
+                propagatedBuildInputs = [
+                  pkgs.SDL_compat
+                  pkgs.SDL_image
+                  pkgs.libwebp
+                ];
+                # when using SDL_compat instead of SDL_classic, SDL_mixer isn't correctly
+                # detected, but there is no harm just specifying it
+                buildInputs = with pkgs; [
+                  git
+                  pkg-config
+                  apple-sdk_15
+
+                  # Need to use brew cmake since nix use old one 3.30.5
+                  # which has issues with copy libraries to macOs app
+                  #cmake
+                  # use clang explicitly since don't use cmake
+                  clang_16
+                  # OpenXcom dependecies
+                  rapidyaml
+                  zlib
+                  #SDL_mixer
+                  # SDL_image
+                  # SDL_gfx
+                  # Try to build using SDL12_compat
+                  (SDL_compat.overrideAttrs (old: {
+                    postInstall = ''
+                      ln -s $out/lib/pkgconfig/sdl12_compat.pc $out/lib/pkgconfig/sdl.pc
```

Comment:
workaround to have working dependecies checking in cmake flow.

Note: The same behavior done in the brew [formula](https://github.com/Homebrew/homebrew-core/blob/master/Formula/s/sdl12-compat.rb#L37)